File: src/models/Project.ts
import mongoose, { Document } from "mongoose";

export interface IProject extends Document {
  userId: string;
  projectId: string;
  containerName: string;
  port?: number;
}

const projectSchema = new mongoose.Schema<IProject>({
  userId: { type: String, required: true, unique: true },
  projectId: { type: String, required: true },
  containerName: { type: String, required: true },
  port: { type: Number, required: false },
});

export const Project = mongoose.model<IProject>("Project", projectSchema);

File: python_version/app/models/project.py
import uuid
from sqlalchemy import Column, String, Integer
from sqlalchemy.dialects.postgresql import UUID
from app.config.database import Base


class Project(Base):
    __tablename__ = "projects"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), unique=True, nullable=False, index=True)
    project_id = Column(String, nullable=False)
    container_name = Column(String, nullable=False)
    port = Column(Integer, nullable=True)
